src/diffusers/pipelines/marigold/pipeline_marigold_depth.py

```diff
@@ -0,0 +1,784 @@
+            image, input_latent, generator, ensemble_size, batch_size
+        )  # [N*E,4,h,w], [N*E,4,h,w]
+
+        del image
+
+        batch_empty_text_embedding = self.empty_text_embedding.to(device=device, dtype=dtype).repeat(
+            batch_size, 1, 1
+        )  # [B,1024,2]
+
+        # 5. Denoising loop. Model invocation: self.unet
+        with self.progress_bar(total=num_images * ensemble_size * denoising_steps) as progress_bar:
+            clean_latent = []
+
+            for i in range(0, num_images * ensemble_size, batch_size):
+                batch_image_latent = image_latent[i : i + batch_size]  # [B,4,h,w]
+                batch_pred_latent = pred_latent[i : i + batch_size]  # [B,4,h,w]
+                B = batch_image_latent.shape[0]
+
+                batch_text_embedding = batch_empty_text_embedding[:B]  # [B,2,1024]
+
+                self.scheduler.set_timesteps(denoising_steps, device=device)
+
+                for t in self.scheduler.timesteps:
+                    batch_latent = torch.cat([batch_image_latent, batch_pred_latent], dim=1)  # [B,8,h,w]
+                    noise = self.unet(batch_latent, t, encoder_hidden_states=batch_text_embedding).sample  # [B,4,h,w]
+                    batch_pred_latent = self.scheduler.step(
+                        noise, t, batch_pred_latent, generator=generator
+                    ).prev_sample  # [B,4,h,w]
+                    progress_bar.update(B)
+
+                clean_latent.append(batch_pred_latent)
+
+                del batch_image_latent, batch_pred_latent, batch_text_embedding, batch_latent, noise
+
+            pred_latent = torch.cat(clean_latent, dim=0)  # [N*E,4,h,w]
+
+            del clean_latent
+
+        del image_latent, batch_empty_text_embedding
+
+        # 6. Decode prediction from latent into pixel space. Model invocation: self.vae.decoder
+        prediction = torch.cat(
+            [
+                self.decode_prediction(pred_latent[i : i + batch_size])
+                for i in range(0, pred_latent.shape[0], batch_size)
+            ],
+            dim=0,
+        )  # [N*E,1,PPH,PPW]
+
+        if not output_latent:
+            pred_latent = None
+
+        # 7. Postprocess predictions
+        prediction = unpad_image(prediction, padding)  # [N*E,1,PH,PW]
+
+        uncertainty = None
+        if ensemble_size > 1:
+            prediction = prediction.reshape(num_images, ensemble_size, *prediction.shape[1:])  # [N,E,1,PH,PW]
+            prediction = [
+                ensemble_depth(prediction[i], output_uncertainty, **(ensembling_kwargs or {}))
+                for i in range(num_images)
+            ]  # [ [[1,1,PH,PW], [1,1,PH,PW]], ... ]
+            prediction, uncertainty = zip(*prediction)  # [[1,1,PH,PW], ... ], [[1,1,PH,PW], ... ]
+            prediction = torch.cat(prediction, dim=0)  # [N,1,PH,PW]
+            uncertainty = torch.cat(uncertainty, dim=0)  # [N,1,PH,PW]
+
+        if match_input_resolution:
```

Comment:
this should be part of `image_processor.post_process`

Reply:
This particular step cannot be moved out, because it is dependent on the pipeline. Specifically, in the normals pipeline there is an extra line `prediction = normalize_normals(prediction)  # [N,3,H,W]` after the resizing, which ensures that the normals are unit-length after resizing with e.g. bilinear interpolation. This ensures validity and cannot be easily moved to postprocessing